src/services/auth.service.ts

```diff
@@ -0,0 +1,49 @@
+declare const gapi: any;
+
+export class AuthService {
+
+  public loadGapi(): void {
+    // Get the gapi script from Google.
+    const gapiScript = document.createElement('script');
+    gapiScript.src = 'https://apis.google.com/js/platform.js';
+    gapiScript.type = 'text/javascript';
+    gapiScript.onload = () => {
+      // Load the specific client libraries we need.
+      this.loadAuth2();
+    };
+    document.head.appendChild(gapiScript);
+  }
+
+  public signIn(): void {
+    gapi.auth2.getAuthInstance().signIn({'prompt': 'select_account'});
+  }
+
+  private loadAuth2(): void {
+    gapi.load('auth2', () => {
+      const accessToken = this.currentAccessToken;
+      if (!accessToken) {
+        // TODO: don't hardcode the client ID, domain; get these from the config
```

Comment:
FWIW, this should already be populated in Leo, so we should only need to specify our own client ID for local development (it may still make sense to have that come from the config).

Reply:
So let's say you sign out of Google in another browser tab while using your notebook... when you try using the notebook again and interact with our extension, it wouldn't work anymore -- we'd need to force a Google sign in at that point. Correct?

If so, that may be a reason to include this in the config for all environments.

Reply:
- I believe Jupyter would not be signed out in that case since Leo uses a custom cookie (https://precisionmedicineinitiative.atlassian.net/browse/RW-202 would change that to match what you're saying).
- Ultimately, Leo needs you to be signed in for Leo to work, so we should be addressing this there rather than in our extension (not sure if that's what you were suggesting). FR: https://github.com/DataBiosphere/leonardo/issues/326
- I agree the client ID should be configured on the cluster; since Leo needs to use it, the most logical place would be a CreateCluster parameter: https://github.com/DataBiosphere/leonardo/issues/289